### ISSUE TITLE
Webpack uses incorrect entry point as of v6.1.1

### ISSUE THREAD

**Issue by msujew** [remojansen/InversifyJS#1616]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I am a maintainer of the Theia IDE framework, and was made aware of this issue in https://github.com/eclipse-theia/theia/discussions/14430. After v6.1.0, webpack no longer correctly find the cjs entry point and instead decides to use the ESM entry point. This eventually results in an obscure webpack error, complaining that `@inversifyjs/core` has no exports at all.

### Steps to reproduce

1. Update inversify to 6.1.1 in [here](https://github.com/eclipse-theia/theia/blob/bc4acd43e5a33dfe64c9bc3fb7ccb6ae9189c76f/packages/core/package.json#L54).
2. Run `yarn && yarn browser build`.
3. Webpack will eventually report the error reported in https://github.com/eclipse-theia/theia/discussions/14430.

### Expected behavior

Webpack continues using the cjs entry point as was the case in 6.0.3

### Possible solution

In my local environment, I've added the following lines of export configuration (which already exist in `@inversifyjs/core`). This seems to fix the issue again:

```json
"exports": {
  ".": {
    "import": "./es/inversify.js",
    "require": "./lib/inversify.js"
  }
}
```

### Package version

6.1.1

### Node.js version

18.20.1

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [X] Linux

### Stack trace

```
WARNING in ../../node_modules/inversify/es/planning/reflection_utils.js 10:22-32
export 'getTargets' (imported as 'getTargets') was not found in '@inversifyjs/core' (module has no exports)
@ ../../node_modules/inversify/es/planning/planner.js 11:0-100 114:31-46 116:47-74 118:69-84
@ ../../node_modules/inversify/es/container/container.js 62:0-85 122:19-39 124:32-52 317:28-45 637:26-30
@ ../../node_modules/inversify/es/inversify.js 4:0-50 4:0-50
@ ./src-gen/backend/server.js 12:22-42
@ ./src-gen/backend/main.js 77:21-40
ERROR in ../../node_modules/@inversifyjs/common/lib/esm/index.js 1:0-80
Module not found: Error: Can't resolve './services/models/LazyServiceIdentifier' in '<project_path>\node_modules@inversifyjs\common\lib\esm'
Did you mean 'LazyServiceIdentifier.js'?
BREAKING CHANGE: The request './services/models/LazyServiceIdentifier' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '.mjs' file, or a '.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
@ ../../node_modules/inversify/es/inversify.js 2:0-60 2:0-60
@ ./src-gen/backend/server.js 12:22-42
@ ./src-gen/backend/main.js 77:21-40
ERROR in ../../node_modules/@inversifyjs/core/lib/esm/index.js 1:0-68
```

**Comment by notaphplover** [comment]
Hey @msujew, thank you so much for the issue report. I had a look at the PR :smiley:

**Comment by notaphplover** [comment]
Ok, I recreated the issue in this [code sandbox](https://codesandbox.io/p/devbox/j59qrk). I'll focus on fixing this issue now, but I think I'm going to add an e2e test so this never happens again.

**Comment by notaphplover** [comment]
Ok I just published  `inversify@6.1.2-beta.1`and updated the [code sandbox](https://codesandbox.io/p/devbox/j59qrk), which can now bundle inversify.

@msujew could you test it targeting this version? If it works in your case as well I'll release a `inversify@6.1.2` version :smiley:. Thank you so much for the report.

**Comment by msujew** [comment]
@notaphplover Thanks a lot for looking into this so quickly! Yes, I can confirm that using the `6.1.2-beta.1` version fixes the reported issue.

**Comment by notaphplover** [comment]
> @notaphplover Thanks a lot for looking into this so quickly! Yes, I can confirm that using the 6.1.2-beta.1 version fixes the reported issue.

My pleasure :). I just released `inversify@6.1.2`. I'll close the issue now. If this error persist, feel free to ping me and I'll reopen it.
